app/services/forest_liana/search_query_builder.rb

```diff
@@ -50,23 +51,25 @@ def search_param
 
         @resource.columns.each_with_index do |column, index|
           column_name = format_column_name(@resource.table_name, column.name)
-          if column.name == 'id'
-            if column.type == :integer
-              value = @params[:search].to_i
-              conditions << "#{@resource.table_name}.id = #{value}" if value > 0
-            elsif REGEX_UUID.match(@params[:search])
-              conditions << "#{@resource.table_name}.id =
-                '#{@params[:search]}'"
+          if (@collection.search_fields && @collection.search_fields.index(column.name)) || !@collection.search_fields
```

Comment:
```ruby
if @collection.search_fields.nil? || (@collection.search_fields && @collection.search_fields.include?(column.name))
```
The "search_fields" condition is almost always `true` so lets make the code more readable and optimise the code execution here.